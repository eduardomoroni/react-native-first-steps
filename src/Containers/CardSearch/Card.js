/* @flow */

import React, { PureComponent } from 'react'
import { View, Text } from 'react-native'
import styles from '../../Styles/CardStyle'
import { placeholdersToSymbols } from '../../Realm/Conversion/Placeholder'

// TODO: Make this component a dumb component
// TODO: Use prop-types
type CardProps = {
  card: any,
  showCardText: boolean
}

export const getLastPrinting = (printings: any) => {
  const keys = Object.keys(printings)
  const lastPrinting = printings[keys[keys.length - 1]]
  return lastPrinting.printing
}

export class Card extends PureComponent {
  props: CardProps

  render () {
    const { card, showCardText } = this.props
    const {
      name,
      text,
      manaCost,
      type,
      power,
      toughness,
      printings
    } = card
    return (
      <View style={styles.container}>
        {renderCardNameAndMana(name, placeholdersToSymbols(manaCost))}
        {renderCardTypeAndEdition(type, getLastPrinting(printings))}
        {renderTextAndPower(text, power, toughness, showCardText)}
      </View>
    )
  }
}

const renderCardNameAndMana = (leftText, rightText) => {
  return (
    <View style={styles.lineContainer}>
      <View style={styles.leftWord}>
        <Text key='cardName' style={styles.cardNameText}>{leftText}</Text>
      </View>
      <View style={styles.rightWord}>
        <Text key='cardManaCost' style={styles.mana}>{rightText}</Text>
      </View>
    </View>
  )
}

const renderCardTypeAndEdition = (leftText, rightText) => {
  return (
    <View style={styles.lineContainer}>
      <View style={styles.leftWord}>
        <Text key='cardType' style={styles.cardTypeText}>{leftText}</Text>
      </View>
      <View style={styles.rightWord}>
        <Text key='cardEdition'>{rightText}</Text>
      </View>
    </View>
  )
}

const renderTextAndPower = (text, power, toughness, showCardText) => {
  return (
    <View style={styles.cardTextContainer}>
      <View style={styles.cardText}>
        <Text key='cardText' style={styles.text}>{showCardText ? text : ''}</Text>
      </View>
      <View style={styles.cardPower}>
        <Text key='cardPowerAndToughness' style={styles.cardPowerToughness}>{power || toughness ? `${power}/${toughness}` : ''}</Text>
      </View>
    </View>
  )
}
