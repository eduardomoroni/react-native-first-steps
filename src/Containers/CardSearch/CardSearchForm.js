/* @flow */

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { View, Keyboard } from 'react-native'
import { reduxForm, Field, formValueSelector } from 'redux-form'
import { searchForCards, showFormModal } from '../../Redux/Actions'
import Styles from '../../Styles/CardSearchFormStyle'
import { valuesOf } from '../../Realm/RealmService'
import type { Dispatch } from 'redux'
import {
  DropdownInputForm,
  Modal,
  MultiSelect,
  ManaIconsBar,
  SubmitButtonForm,
  NumericInputForm,
  ModalToggle,
  TextInputForm
} from '../Components'

export class CardSearchForm extends Component {
  renderThreeFieldInRow (fieldOne, fieldTwo, fieldThree) {
    return (
      <View style={Styles.multipleFieldsPerLine}>
        <View style={Styles.leftField}>
          {fieldOne}
        </View>
        <View style={Styles.middleField}>
          {fieldTwo}
        </View>
        <View style={Styles.rightField}>
          {fieldThree}
        </View>
      </View>
    )
  }

  renderTwoFieldInRow (leftField, rightField) {
    return (
      <View style={Styles.multipleFieldsPerLine}>
        <View style={Styles.leftField}>
          {leftField}
        </View>
        <View style={Styles.rightField}>
          {rightField}
        </View>
      </View>
    )
  }

  renderModal () {
    let modalContent = {}
    let { visibleModal, cardSets, cardFormats, cardRarities } = this.props

    if (visibleModal === 'cardRarity') {
      modalContent = <Field name='cardRarity' component={MultiSelect} items={cardRarities} />
    } else if (visibleModal === 'cardSet') {
      modalContent = <Field name='cardSet' component={MultiSelect} items={cardSets} />
    } else if (visibleModal === 'cardFormat') {
      modalContent = <Field name='cardFormat' component={MultiSelect} items={cardFormats} />
    } else {
      return <View />
    }
    return (
      <Modal isVisible onModalHide={() => this.props.showModal('')} >
        {modalContent}
      </Modal>
    )
  }

  render () {
    const {
      handleSubmit,
      cardTypes,
      cardSubTypes,
      cardType,
      cardSubType,
      showModal,
      searchCards,
      cardRarity,
      cardSet,
      cardFormat
    } = this.props

    const formSubmit = (formValues) => {
      Keyboard.dismiss()
      console.log('Form sent: ', formValues)
      searchCards(formValues)
    }

    const numericOperators = ['', '<', '<=', '=', '>=', '=']

    return (
      <View style={Styles.container}>
        <View style={Styles.formContainer}>
          <Field name='cardName' component={TextInputForm} />
          <Field name='cardText' component={TextInputForm} />
          { this.renderTwoFieldInRow(
            <Field name='cardType' component={DropdownInputForm} dropdownItems={cardTypes} selectedValue={cardType} />,
            <Field name='cardSubType' component={DropdownInputForm} dropdownItems={cardSubTypes} selectedValue={cardSubType} />
        )}
          { this.renderThreeFieldInRow(
            <Field name='cardPower' component={NumericInputForm} dropdownItems={numericOperators} />,
            <Field name='cardToughness' component={NumericInputForm} dropdownItems={numericOperators} />,
            <Field name='cardCMC' component={NumericInputForm} dropdownItems={numericOperators} />
        )}
          <Field name='cardColors' component={ManaIconsBar} />
          {/* Secundary Options */}
          <Field name='cardFlavorText' component={TextInputForm} />
          { this.renderTwoFieldInRow(
            <Field name='cardArtist' component={TextInputForm} />,
            <Field name='cardCollectionNumber' component={TextInputForm} keyboardType={'numeric'} maxLength={3} />
        )}
          { this.renderThreeFieldInRow(
            <ModalToggle label='cardRarity' onPress={showModal} selected={cardRarity} />,
            <ModalToggle label='cardSet' onPress={showModal} selected={cardSet} />,
            <ModalToggle label='cardFormat' onPress={showModal} selected={cardFormat} />
        )}
          <Field name='cardColorsIdentity' component={ManaIconsBar} />
          {this.renderModal()}
        </View>
        <View style={Styles.containerFooter}>
          <SubmitButtonForm onPress={handleSubmit(formSubmit)} />
        </View>
      </View>
    )
  }
}

Field.propTypes = {
  dropdownItems: PropTypes.arrayOf(PropTypes.string),
  selectedValue: PropTypes.string
}

const mapStateToProps = (state) => {
  const selector = formValueSelector('CardSearchForm')

  const cardRarities = ['Common', 'Uncommon', 'Rare', 'Mythic Rare']
  const cardTypes = valuesOf('Type')
  const cardSubtypes = valuesOf('SubType')
  const printings = valuesOf('Printing')
  const formats = valuesOf('Legality')

  cardTypes.unshift('')
  cardSubtypes.unshift('')

  return {
    cardType: selector(state, 'cardType'),
    cardSubType: selector(state, 'cardSubType'),
    cardSubTypes: cardSubtypes,
    cardTypes: cardTypes,
    cardSets: printings,
    cardFormats: formats,
    cardRarity: selector(state, 'cardRarity'),
    cardSet: selector(state, 'cardSet'),
    cardFormat: selector(state, 'cardFormat'),
    visibleModal: state.cardSearch.visibleModal,
    cardRarities: cardRarities
  }
}

const mapDispatchToProps = (dispatch: Dispatch<*>) => {
  return {
    searchCards: (cardSearchForm) => dispatch(searchForCards(cardSearchForm)),
    showModal: (modal) => dispatch(showFormModal(modal))
  }
}

const cardSearchFormDecorated = reduxForm({form: 'CardSearchForm'})(CardSearchForm)
export default connect(mapStateToProps, mapDispatchToProps)(cardSearchFormDecorated)
