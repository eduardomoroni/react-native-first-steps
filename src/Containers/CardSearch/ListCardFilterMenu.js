// @flow

import React, { Component } from 'react'
import { View, BackAndroid, Switch, Text, Picker } from 'react-native'
import { reduxForm, Field, formValueSelector } from 'redux-form'
import { connect } from 'react-redux'
import { toggleShowCardText, sortCards as sortCardsActionCreator } from '../../Redux/Actions'
import I18n from 'react-native-i18n'

const renderSwitch = (fieldProps) => {
  const { input, isEnabled } = fieldProps
  const { onChange, name } = input

  return (
    <View style={styles.rowStyle}>
      <Text>{I18n.t(name)}</Text>
      <Switch style={styles.switchStyle} value={isEnabled} onValueChange={onChange} />
    </View>
  )
}

const renderSortInput = (fieldProps) => {
  const { onChange, name, value } = fieldProps.input

  const sortValues = ['cmc', 'multiverseid', 'name', 'number', 'rarity', 'type', 'power', 'toughness']

  const renderPickerItem = (value, key) => {
    return <Picker.Item label={value} value={value} key={key} />
  }

  return (
    <View style={styles.rowStyle}>
      <Text>{I18n.t(name)}</Text>
      <Picker
        selectedValue={value.field}
        onValueChange={onChange}>
        { sortValues.map(renderPickerItem) }
      </Picker>
    </View>
  )
}

class ListCardFilterMenu extends Component {
  componentDidMount () {
    BackAndroid.addEventListener('hardwareBackPress', () => {
      if (this.context.drawer.props.open) {
        this.context.drawer.toggle()
        return true
      }
      return false
    })
  }

  componentWillReceiveProps (nextProps) {
    const {
      showCardText,
      toggleCardText,
      sortBy,
      sortCards,
      sortReverseOrder
    } = nextProps

    if (this.props.showCardText !== showCardText) {
      toggleCardText({showCardText: showCardText})
    }
    if (this.props.sortBy !== sortBy || this.props.sortReverseOrder !== sortReverseOrder) {
      sortCards({field: sortBy, reversed: sortReverseOrder})
    }
  }

  render () {
    const { showCardText, sortReverseOrder } = this.props
    return (
      <View style={styles.container}>
        <Field name='showCardText' component={renderSwitch} isEnabled={showCardText} />
        <Field name='sortReverseOrder' component={renderSwitch} isEnabled={sortReverseOrder} />
        <Field name='sortBy' component={renderSortInput} />
      </View>
    )
  }
}

ListCardFilterMenu.contextTypes = {
  drawer: React.PropTypes.object
}

ListCardFilterMenu.propTypes = {
  showCardText: React.PropTypes.bool,
  sortBy: React.PropTypes.string,
  sortReverseOrder: React.PropTypes.bool,
  toggleCardText: React.PropTypes.func,
  sortCards: React.PropTypes.func
}

Field.propTypes = {
  isEnabled: React.PropTypes.bool
}

const mapStateToProps = (state) => {
  const selector = formValueSelector('CardSearchFilter')

  return {
    initialValues: {
      showCardText: state.cardSearch.showCardText,
      sortBy: state.cardSearch.sortBy.field,
      sortReverseOrder: state.cardSearch.sortBy.reversed
    },
    showCardText: selector(state, 'showCardText'),
    sortBy: selector(state, 'sortBy'),
    sortReverseOrder: selector(state, 'sortReverseOrder')
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    toggleCardText: (filter) => dispatch(toggleShowCardText(filter)),
    sortCards: (sortParams) => dispatch(sortCardsActionCreator(sortParams))
  }
}

const styles = {
  container: {
    flex: 1,
    padding: 20
  },
  switchStyle: {
    marginBottom: 10
  },
  rowStyle: {
    height: 40
  }
}

const ListCardFilterDecorated = reduxForm({form: 'CardSearchFilter'}, mapStateToProps)(ListCardFilterMenu)
export default connect(mapStateToProps, mapDispatchToProps)(ListCardFilterDecorated)
