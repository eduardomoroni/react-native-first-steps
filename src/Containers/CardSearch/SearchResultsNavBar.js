import React, { Component } from 'react'
import { formValueSelector } from 'redux-form'
import { connect } from 'react-redux'
import { navBackButton, navTitle, navButtonBar, openDrawer } from '../../Navigation/NavBarItems'
import { switchDisplayMode as switchDisplayModeAction } from '../../Redux/Actions'
import Menu from '../CardSearch/ListCardFilterMenu'
import CustomNavBar from '../../Navigation/CustomNavBar'

class SearchResultsNavBar extends Component {
  render () {
    const { showCardsAs, switchDisplayMode } = this.props

    const settingsButton = {
      buttonProps: {name: 'sliders'},
      callback: () => openDrawer(<Menu />)
    }

    const displayModeButton = {
      buttonProps: {name: showCardsAs === 'image' ? 'th-list' : 'th-large'},
      callback: () => switchDisplayMode()
    }

    return (
      <CustomNavBar
        leftRender={navBackButton()}
        middleRender={navTitle('searching_cards')}
        rightRender={navButtonBar([displayModeButton, settingsButton])} />
    )
  }
}

const mapStateToProps = (state) => {
  const selector = formValueSelector('CardSearchFilter')
  return {
    showCardsAs: selector(state, 'showCardsAs')
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    switchDisplayMode: () => dispatch(switchDisplayModeAction())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(SearchResultsNavBar)
