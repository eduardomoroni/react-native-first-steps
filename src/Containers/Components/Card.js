/* @flow */

import React, { Component } from 'react'
import {
  View,
  Text,
  StyleSheet
} from 'react-native'

type CardProps = {
  card: any
}

export default class Card extends Component {
  props: CardProps
  render () {
    const {
      name,
      text
    } = this.props.card
    return (
      <View style={styles.container}>
        <Text>{name}</Text>
        <Text>{text}</Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  }
})
