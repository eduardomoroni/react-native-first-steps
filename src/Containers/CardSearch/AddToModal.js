import React, { Component } from 'react'
import { View, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'
import { cardType } from '../../Types/CardType'
import { Modal, SubmitButtonForm, DropdownInputForm } from '../Components'
import WishListService from '../../Services/WishListService'
import { reduxForm, Field, formValueSelector } from 'redux-form'
import { connect } from 'react-redux'

import { RkText } from 'react-native-ui-kitten'

export class AddToModal extends Component {
  constructor () {
    super()
    this.state = {
      isVisible: true
    }
  }

  toggleModal () {
    this.state.isVisible = !this.state.isVisible
  }

  addCardAmountToList (listName, card, amount) {
    const { wishList } = this.props
    console.log(wishList[listName], listName, amount)
    WishListService.updateCardAmount(card, 2, wishList.have)
  }

  render () {
    const {
      card,
      wishList,
      howMuchCards,
      whichTradeList
    } = this.props

    const tradeLists = ['want', 'have']
    const numberRange = ['1', '2', '3', '4']

    console.log(wishList)
    return (
      <Modal isVisible onModalHide={() => this.toggleModal()} >
        <View style={styles.modal}>
          <RkText style={styles.title}>
            {`Add to TradeList`}
          </RkText>
          <Field name='whichTradeList' component={DropdownInputForm} dropdownItems={tradeLists} selectedValue={whichTradeList} />
          <Field name='howMuchCards' component={DropdownInputForm} dropdownItems={numberRange} selectedValue={howMuchCards} />
        </View>
        <View>
          <SubmitButtonForm onPress={() => this.addCardAmountToList(whichTradeList, card, howMuchCards)} />
        </View>
      </Modal>
    )
  }
}

AddToModal.propTypes = {
  card: cardType.isRequired,
  listToAdd: PropTypes.any
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  title: {
    fontSize: 18,
    textAlign: 'center'
  },
  modal: {
    backgroundColor: 'white',
    height: 400,
    borderTopRightRadius: 4,
    borderTopLeftRadius: 4
  }
})

const mapStateToProps = (state) => {
  const selector = formValueSelector('AddToModalForm')

  return {
    whichTradeList: selector(state, 'whichTradeList'),
    howMuchCards: selector(state, 'howMuchCards')
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
  }
}

const AddToModalDecorated = reduxForm({form: 'AddToModalForm'})(AddToModal)
export default connect(mapStateToProps, mapDispatchToProps)(AddToModalDecorated)
