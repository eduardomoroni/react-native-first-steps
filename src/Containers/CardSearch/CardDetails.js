/* @flow */

import React, { Component } from 'react'
import { View, StyleSheet } from 'react-native'
import { Metrics } from '../../Styles/Themes'
import { CardImage } from '../Components'

type CardDetailsProps = {
  card: any
}

export default class CardDetails extends Component {
  props: CardDetailsProps

  render () {
    return (
      <View style={styles.container}>
        <CardImage card={this.props.card} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: Metrics.navBarHeight
  }
})
