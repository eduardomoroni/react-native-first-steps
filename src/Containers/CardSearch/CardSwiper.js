import React, { Component } from 'react'
import PropTypes from 'prop-types'
import CardDetails from './CardDetails'
import { Actions as NavigationActions } from 'react-native-router-flux'
import { cardType } from '../../Types/CardType'
import { getIndex } from '../../Realm/RealmService'
import Swiper from 'react-native-swiper'

class CardSwiper extends Component {
  renderCardDetails (card, key) {
    return (
      <CardDetails card={card} key={key} />
    )
  }

  updateScreenTitle (e, state) {
    const { index } = state
    const getCardName = (index) => this.children[index].props.card.name
    NavigationActions.refresh({title: `${getCardName(index)}`})
  }

  render () {
    const { cards, card } = this.props
    let { index } = this.props

    if (card !== null) {
      index = getIndex(cards, card)
    }

    return (
      <Swiper
        ref={(component) => { this.swiper = component }}
        onMomentumScrollEnd={this.updateScreenTitle}
        style={{ flex: 1 }}
        loadMinimal
        loadMinimalSize={2}
        loop={false}
        showsPagination={false}
        index={index}
        >
        {cards.map(this.renderCardDetails)}
      </Swiper>
    )
  }
}

CardSwiper.propTypes = {
  cards: PropTypes.objectOf(cardType),
  card: cardType,
  index: PropTypes.number
}

CardSwiper.defaultProps = {
  index: 0,
  card: null
}

export default CardSwiper
