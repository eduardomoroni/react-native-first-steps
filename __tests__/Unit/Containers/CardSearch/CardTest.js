import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import { Card, getLastPrinting } from '../../../../src/Containers/CardSearch/Card'
import * as Convertion from '../../../../src/Realm/Conversion/Placeholder'

const props = {
  showCardText: true,
  card: {
    name: 'CardName',
    text: 'CardText',
    manaCost: '{1}{W}{B}',
    type: 'CardType',
    power: 2,
    toughness: 4,
    printings: { '0': { printing: 'AER' } }
  }
}

const hasKey = key => {
  return n => n.key() === key
}

const findRenderedTextFromKey = (wrapper, key) => {
  return wrapper.findWhere(hasKey(key)).childAt(0).text()
}

describe('<Card />', () => {
  const {
    name,
    manaCost,
    type,
    text,
    power,
    toughness,
    printings
  } = props.card
  Convertion.placeholdersToSymbols = jest.fn(text => `placeholdersToSymbols(${text})`)

  it('should render Card component', () => {
    const wrapper = shallow(<Card {...props} />)
    expect(wrapper).toMatchSnapshot()
    expect(findRenderedTextFromKey(wrapper, 'cardName')).toEqual(name)
    expect(findRenderedTextFromKey(wrapper, 'cardManaCost')).toEqual(Convertion.placeholdersToSymbols(manaCost))
    expect(findRenderedTextFromKey(wrapper, 'cardType')).toEqual(type)
    expect(findRenderedTextFromKey(wrapper, 'cardEdition')).toEqual(getLastPrinting(printings))
    expect(findRenderedTextFromKey(wrapper, 'cardText')).toEqual(text)
    expect(findRenderedTextFromKey(wrapper, 'cardPowerAndToughness')).toEqual(`${power}/${toughness}`)
    expect(Convertion.placeholdersToSymbols).toHaveBeenCalledWith(props.card.manaCost)
  })

  it('should hide card text', () => {
    const wrapper = shallow(<Card {...props} showCardText={false} />)
    expect(findRenderedTextFromKey(wrapper, 'cardText')).toEqual('')

    wrapper.setProps({...props, showCardText: true})
    expect(findRenderedTextFromKey(wrapper, 'cardText')).toEqual(props.card.text)
  })

  it('should show only the lastest printing as cardEdition', () => {
    const printExamples = { '0': { printing: 'STH' }, '1': { printing: 'TPR' }, '2': { printing: 'EXP' } }
    const wrapper = shallow(<Card card={{printings: printExamples}} />)
    expect(findRenderedTextFromKey(wrapper, 'cardEdition')).toEqual('EXP')
  })
  it('should extract last printing from Realm representation', () => {
    const realmRepresentation = { '0': { printing: 'STH' }, '1': { printing: 'TPR' }, '2': { printing: 'AER' } }
    expect(getLastPrinting(realmRepresentation)).toEqual('AER')
  })
})
