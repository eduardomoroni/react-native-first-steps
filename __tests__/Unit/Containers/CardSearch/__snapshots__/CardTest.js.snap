// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render Card component 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontFamily": "beleren",
            "fontSize": 17,
          }
        }
      >
        CardName
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontFamily": "MTGX",
            "fontSize": 12,
          }
        }
      >
        placeholdersToSymbols({1}{W}{B})
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontFamily": "MTGX",
            "fontSize": 14,
          }
        }
      >
        CardType
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        AER
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "paddingTop": 6,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontFamily": "MTGX",
            "fontSize": 17,
          }
        }
      >
        CardText
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "column-reverse",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontFamily": "MTGX",
            "fontSize": 17,
          }
        }
      >
        2/4
      </Text>
    </View>
  </View>
</View>
`;
